### PR TITLE
LPS-188208 Force to set class in AssertFailure method

### PR DESCRIPTION
**Pull request by murilostodolni**
Related: https://liferay.atlassian.net/browse/LPS-188208

### CODE REVIEW

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java [hunk 1]
```diff
@@ -823,132 +826,130 @@ public void testAddObjectEntry() throws Exception {
...
-			_user.getUserId(), accountEntry2.getAccountEntryGroupId(),
-			_buyerRole.getRoleId());
+		_deleteAccountEntryOrganizationRel(accountEntry1, suborganization1);
 
-		Assert.assertNotNull(_addObjectEntry(accountEntry1));
+		_removeResourcePermission(
+			ObjectActionKeys.ADD_OBJECT_ENTRY, _accountManagerRole);
 
-		// Organization scope
+		// Account scope permission
```
Comment (seltong) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:896]: Can we sort this by comment?

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java [hunk 2]
```diff
@@ -2245,6 +2215,14 @@ private AccountEntry _addAccountEntry() throws Exception {
 			ServiceContextTestUtil.getServiceContext());
 	}
 
+	private void _addAccountEntryOrganizationRel(
+			AccountEntry accountEntry, Organization organization)
+		throws Exception {
```
Comment (seltong) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:2220]: What do you think about changing the method signature to 

```
private void _addAccountEntryOrganizationRel(
			long accountEntryId, long organizationId)
		throws Exception {
```
?

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java [hunk 3]
```diff
@@ -2736,6 +2714,16 @@ private ObjectFieldSetting _createObjectFieldSetting(
 		return objectFieldSetting;
 	}
 
+	private void _deleteAccountEntryOrganizationRel(
```
Comment (seltong) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:2717]: The same case of `_addAccountEntryOrganizationRel`

**Review by seltong — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java (hunk 2 above)
Comment (murilostodolni) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:2220]: Hey @seltong

If we change to  `long accountEntryId, long organizationId` we will need two lines to call the method. I prefer pass all entity an use just one line in call. See same [here](https://github.com/liferay/liferay-portal/blob/53891c022ac74ff0a55e3a6da7dc738340966266/modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java#L2288).

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java (hunk 1 above)
Comment (murilostodolni) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:896]: I don't understood.

Comment (seltong) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:896]: We talked and decided to keep it that way.

File: portal-test/src/com/liferay/portal/kernel/test/AssertUtils.java [hunk 4]
```diff
@@ -127,20 +127,11 @@ public static void assertFailure(
 			Assert.fail();
 		}
 		catch (Exception exception) {
-			if (clazz != null) {
```
Comment (seltong) [inline portal-test/src/com/liferay/portal/kernel/test/AssertUtils.java:130]: Hey, could you please check if this change is related [to](https://testray.liferay.com/home/-/testray/case_results/2357586883?redirect=https%3A%2F%2Ftestray.liferay.com%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D2357582358%26cur%3D1%26delta%3D200%26tab%3Druns%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc) 

```
PACKAGE_NAME                                       DELTA      CUR_VER    BASE_VER   REC_VER    WARNINGS  
[beanshell] = ================================================== ========== ========== ========== ========== ==========
[beanshell] * com.liferay.portal.kernel.test                     MAJOR      2.4.0      2.4.0      3.0.0      VERSION INCREASE REQUIRED
[beanshell] 	>   class      com.liferay.portal.kernel.test.AssertUtils
[beanshell] 		-   method     assertFailure(java.lang.String,com.liferay.petra.function.UnsafeSupplier)
[beanshell] 			-   access     static
[beanshell] 			-   return     void
[beanshell] [Baseline Warning] Bundle Version Change Recommended: 17.0.0
```

**Comment by murilostodolni** [comment]
ci:test:sf

**Comment by murilostodolni** [comment]
ci:test:relevant

**Comment by murilostodolni** [comment]
ci:test:object

**Review by seltong — APPROVED** [review]

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java [hunk 5]
```diff
@@ -823,132 +826,130 @@ public void testAddObjectEntry() throws Exception {
...
 				"User ", _user.getUserId(),
-				" must have ADD_OBJECT_ENTRY permission for ",
-				_objectDefinition3.getResourceName(), StringPool.SPACE),
-			() -> _addObjectEntry(accountEntry1));
+				" does not have access to account entry ",
+				accountEntry2.getAccountEntryId()),
+			() -> _addObjectEntry(accountEntry2));
 
-		_addResourcePermission("ADD_OBJECT_ENTRY", _buyerRole);
+		// Organization account entry restricted scope
```
Comment (carolmariaabb) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:848]: @murilostodolni can we rename it to `Account entry restricted scope with organization`?

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java [hunk 6]
```diff
@@ -2220,6 +2188,8 @@ public void testUpdateObjectEntryAccountRestriction() throws Exception {
 			objectEntry1.getId(), objectEntry1);
 
 		AssertUtils.assertFailure(
+			ObjectEntryValuesException.UnmodifiableAccountEntryObjectField.
+				class,
```
[the remark this answers is not in the document]
Comment (murilostodolni) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:2192]: Perfect, I will sort this!

**Comment by murilostodolni** [comment]
ci:test:sf

**Comment by murilostodolni** [comment]
ci:test:relevant

**Comment by murilostodolni** [comment]
ci:test:object

**Comment by murilostodolni** [comment]
ci:test:sf

**Comment by murilostodolni** [comment]
ci:test:relevant

**Comment by murilostodolni** [comment]
ci:test:object

**Review by carolmariaabb — APPROVED** [review]
LGTM!

**Comment by murilostodolni** [comment]
ci:test:relevant

**Comment by murilostodolni** [comment]
ci:test:object

**Comment by carolmariaabb** [comment]
ci:forward:force

**Pull request closed without merging**